db_stress_tool/db_stress_test_base.cc

```diff
@@ -2165,5 +2170,76 @@ void StressTest::Reopen(ThreadState* thread) {
           num_times_reopened_);
   Open();
 }
+
+void StressTest::MaybeStartTrace() {
+#ifndef ROCKSDB_LITE
+  if (FLAGS_trace_path.empty()) {
+    return;
+  }
+  trace_enabled_ = true;
+
+  if (FLAGS_use_txn) {
+    fprintf(stdout, "--trace_path with --use_txn is not supported.\n");
+    trace_enabled_ = false;
+  }
+
+  if (FLAGS_use_blob_db) {
+    fprintf(stdout, "--trace_path with --use_blob_db is not supported.\n");
+    trace_enabled_ = false;
+  }
+
+  if (trace_env == nullptr) {
+    fprintf(stdout, "trace_env is null.  Disable --trace.\n");
+    trace_enabled_ = false;
+  }
+
+  if (trace_enabled_ == false) {
+    fprintf(stdout, "--trace_path is disabled.\n");
+    return;
+  }
+
+  EnvOptions trace_env_options(options_);
+  std::unique_ptr<TraceWriter> trace_writer;
+
+  Status s = NewFileTraceWriter(
+      trace_env, trace_env_options, FLAGS_trace_path, &trace_writer);
+  if (!s.ok()) {
+    trace_enabled_ = false;
+    fprintf(stdout, "NewFileTraceWriter() failed -- %s\n",
+        s.ToString().c_str());
+    return;
+  }
+
+  s = db_->StartTrace(TraceOptions(), std::move(trace_writer));
+  if (!s.ok()) {
+    trace_enabled_ = false;
+    fprintf(stdout, "StartTrace() failed -- %s\n",
```

Comment:
Are these failure/incompatibility messages stderr worthy?

Reply:
Do you mean moving from stdout to stderr?  Or completely remove the message?

Reply:
> That's interesting, I guess we want db_stress to fail if it's configured to enable trace file and the limit is exceeded. Also how does the limit work when the file is closed and reopened?

So it will stop tracing new writes but keep the existing (earlier) trace.

Reply:
I meant moving from stdout to stderr.